src/pay-period-calendar/pay-period-selection.model.ts

```diff
@@ -0,0 +1,6 @@
+import { PayPeriod }                from './pay-period.model';
+
+export interface PayPeriodSelection {
+    payPeriod: PayPeriod;
+    dayIndex: number;
```

Comment:
at some point we'll enable strict null checks, so let's make `dayIndex` be `number | null`.